### DISCUSSION TITLE
Performance NextJS 13.2.4 vs 14.0.2 - self hosting

### DISCUSSION THREAD

**Discussion by pawelmidur** [vercel/next.js#58676] [Help]
### Summary

Hi, 
I have a problem with stability and performance after upgrading next. Nothing other than the update has been added to the application, so I am ruling out other factors.

I use Next in PAGE Router. 

On version: NextJS v13.2.4 (NodeJS - v18.16.0)
Server: 24GB RAM, 16 GPU (VM on GCP), 
Runtime: pm2, exec_mode: "cluster" - 6 instances
Traffic of ~4000-5000 users (real time on analytics google),
Server usage ~5BG RAM and CPU average ~50%,
Locust Test: 0% errors, 0% error: 500, ~160request/s (a whole month)

On version: NextJS v14.0.2 (NodeJS - v20.9.0)
Server: 24GB RAM, 16 GPU (VM on GCP), 
Runtime: pm2, exec_mode: "cluster" - 6 instances
Traffic of ~4000-5000 users (real time on analytics google),
Server usage ~7BG RAM and CPU average ~90%,
Locust Test: 1% errors, 5% error: 500, ~170request/s (every test)

On version: NextJS v14.0.2 (NodeJS - v20.9.0)
Server: 24GB RAM, 24 GPU (VM on GCP), 
Runtime: pm2, exec_mode: "cluster" - 8 instances
Traffic of ~4000-5000 users (real time on analytics google),
Server usage ~7BG RAM and CPU average ~60-70%,
Locust Test: 1% errors, 2% error: 500, ~180request/s

What could be the reason for such a large number of occurrences of pages with error 500? with the same traffic
Nothing has changed in the application itself

Another problem that occurs is the very long first call, on the previous version there was not this problem,

Has anyone had a similar problem? Does anyone have any tips?

I know it's hard to advise something when you don't have a lot of details, but any advice will be valuable

### Additional information

```js
current pm2 file:

...
  "script": "node_modules/next/dist/bin/next",
  "args": "start",
  "instances": 8,
  "exec_mode": "cluster",
  "wait_ready": true,
  "watch": false,
  "autorestart": true,
...
```

```
node_modules/next/dist/bin/next info

Operating System:
  Platform: linux
  Arch: x64
  Version: #20~22.04.1-Ubuntu SMP Mon Oct 23 12:29:43 UTC 2023
Binaries:
  Node: 20.9.0
  npm: 10.1.0
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.0.2
  eslint-config-next: 14.0.2
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.1.6
Next.js Config:
  output: N/A
```
```


### Example

_No response_

**Comment by pawelmidur** [comment]
Initial server response time - 1,720 ms.... earlier before updating ~300ms

**Comment by AdamZajler** [comment]
I'm having same issue here https://github.com/vercel/next.js/discussions/58822
